service/src/main/kotlin/xyz/lebkuchenfm/external/youtube/YoutubeClient.kt

```diff
@@ -0,0 +1,56 @@
+package xyz.lebkuchenfm.external.youtube
+
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.engine.okhttp.OkHttp
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.get
+import io.ktor.client.request.parameter
+import io.ktor.http.parameters
+import io.ktor.serialization.kotlinx.json.json
+import io.ktor.server.config.ApplicationConfig
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+
+class YoutubeClient(config: ApplicationConfig) {
+    private val youtubeClient: HttpClient by lazy { prepareHttpClient() }
+    private val apiKey: String by lazy { config.property("youtube.apiKey").getString() }
+
+    suspend fun getVideoName(id: String): String {
+        val response: YoutubeVideosResponse =
+            youtubeClient.get("/videos") {
+                parameters {
+                    parameter("id", id)
+                    parameter("part", "snippet")
+                    parameter("filer", "id")
+                }
+            }.body()
+
+        val video = response.items.first()
```

Comment:
We will have to come back to these clients and handle errors properly. I will make a note of this in our document.